### DISCUSSION TITLE
When I update an image in the Tiptap editor, the image extension changes.

### DISCUSSION THREAD

**Discussion by ekremhuseyinhancer** [awcodes/filament-tiptap-editor#483] [Q&A] [answered]
Hello, when I try to update an image in the Tiptap editor, the image disappears.
The initial code snippet I used: 
`<img src="/storage/footer/5a14d7f8-30ea-4b37-8995-03dd079566e2.png" alt="asd" title="asd" width="1000" height="500">`
It seems like there's an error in the code after updating the image:
`<img src="/footer/5a14d7f8-30ea-4b37-8995-03dd079566e2.png" alt="asd" title="asd" width="1000" height="500">
`What can I do to fix this issue? Thanks in advance for any assistance :)
<img width="1440" alt="Ekran Resmi 2024-10-15 13 48 17" src="https://github.com/user-attachments/assets/1375a004-281d-45a2-a8da-ae021dedc73d">
<img width="1440" alt="Ekran Resmi 2024-10-15 13 48 09" src="https://github.com/user-attachments/assets/a3ee5346-a14c-4475-ae9f-1a380736afe1">

**Comment by awcodes** [comment]
Looks like you are loosing the 'storage' part of the path. I can't replicate this on my demo.

Can you provide a reproduction repo?

**Reply by ohiader** [reply]
I'm only able to reproduce this on subdomains. It looks like when `use_relative_paths` is true - updating an image (without replacing it) removes `/storage` from the url. My quick and dirty fix is to add this to the related if section in `MediaAction.php` and `EditMediaAction.php`'s ->action().

```php
if (config('filament-tiptap-editor.disk') === 'public' && ! Str::startsWith($source, '/storage/')) {
    $source = '/storage'.$source;
}
```

It looks like `$data['src']` is losing `/storage` in both files - but `$arguments['src']` keeps it, at least in `EditMediaAction.php`.

**Reply by awcodes** [reply]
Hmm. Looks like config('app.url') is the culprit. I'll look into further as soon as I can. Thanks for the further debugging.

**Reply by awcodes** [reply]
I'm not seeing anything in the code that would cause it to loose the '/storage' segment.

Can you provide a reproduction repo? I don't have anything setup with subdomains to debug further.

**Reply by fedenuche** [reply, marked as answer]
I had the same problem. In my case, I fixed that just by changing the `'use_relative_paths' => false` in the filament-tiptap-editor.php config file.

**Reply by ekremhuseyinhancer** [reply]
First of all, I want to thank all of you. I was about to finish the last part of my project. The project was almost completed. Thanks to fedenuche, my problem was solved. I thank you all very much for your efforts.
